docker-app/qfieldcloud/core/serializers.py

```diff
@@ -95,6 +96,29 @@ def validate(self, data):
 
         return data
 
+    def get_localized_datasets(self, obj):
+        """
+        Extracts all localized datasets from `project_details["layers_by_id"]`.
+        A localized dataset is identified by `is_localized: true`.
+        """
+        if not obj.project_details or "layers_by_id" not in obj.project_details:
+            return []
+
+        layers = obj.project_details.get("layers_by_id", {})
+
+        localized_datasets = [
+            {
+                "id": layer.get("id"),
+                "name": layer.get("name"),
+                "datasource": layer.get("datasource"),
+                "filename": layer.get("filename"),
+            }
+            for layer in layers.values()
+            if layer.get("is_localized", False)
+        ]
+
+        return localized_datasets
```

Comment:
```suggestion
        return filter(lambda ld: ld.get("is_localized", False),  layers.values())
```

Two suggestions:
- we avoid renaming the `layers_by_id` dictionary by simply returning those
- we slighthy pre-maturely optimize with a filter object instead of a list

Reply:
Thanks for the suggestions:

- Moved the method to the Project model as proposed
- Used `filter()` with lambda to have a slight optimization
- Thanks for pointing this out, it's a great trade-off to consider, list or filter. My propose is to keep the list as it works better with serializers and we avoid any suprises.

Reply:
@tdrivas can we close this comment?